File: web/src/app/filmes/services/models/filme-output.ts
import {GeneroOutput} from "../../../generos/services/models/genero-output";
import {AtorOutput} from "../../../atores/services/models/ator-output";
import {CinemaOutput} from "../../../cinemas/services/models/cinema-output";

export interface FilmeOutput {
  id: string;
  nome: string;
  noCinema: boolean;
  dataDeLancamento: Date;
  foto: string | null;
  generos: GeneroOutput[];
  atores: AtorOutput[];
  cinemas: CinemaOutput[];
}

File: server/filmpalace/src/main/java/com/rossetti/filmpalace/dto/FilmeOutput.java
package com.rossetti.filmpalace.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotNull;
import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
public class FilmeOutput {
    @NotNull
    private UUID id;
    @NotNull
    private String nome;
    @NotNull
    private Boolean noCinema;
    @NotNull
    private LocalDate dataDeLancamento;
    @NotNull
    private String foto;
    @NotNull
    private List<GeneroOutput> generos;
    @NotNull
    private List<AtorOutput> atores;
    @NotNull
    private List<CinemaOutput> cinemas;
}
